commands/operator-sdk/cmd/scorecard/test_definitions_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import "testing"
+
+// TestSuiteWeightsCheck makes sure that the combined weights of all
+// the tests in a suite adds up to 100
+func TestSuiteWeightsCheck(t *testing.T) {
+	basicTests := NewBasicTestSuite(BasicTestConfig{})
+	basicWeight := 0
+	for _, weight := range basicTests.Weights {
+		basicWeight += weight
+	}
+	if basicWeight != 100 {
+		t.Errorf("Weights of Basic Tests add to %d", basicWeight)
+	}
+	olmTests := NewOLMTestSuite(OLMTestConfig{})
+	OLMWeight := 0
+	for _, weight := range olmTests.Weights {
+		OLMWeight += weight
+	}
+	if OLMWeight != 100 {
+		t.Errorf("Weights of OLM Tests add to %d", OLMWeight)
+	}
+}
```

Comment:
I can't decide if it would be more or less intuitive for the weights to be relative. For example, I could have three tests, with weights [2,1,1] which would be equivalent to [50%,25%,25%].

Also, relative weights might be necessary if the test suite weights don't evenly divide into 100. For example, it's not possible to have 3 tests with equal weights that add to 100. The closest approximation would be [33,33,34], and that would be difficult to do if tests are being added to a test suite in a loop. Floating point weights could also solve that problem.

Reply:
I think that's a good idea. It would make it much easier to configure the scores the way we want them. This was already a problem because `CRDsHaveValidation` was supposed to have 25% more weight than the other OLM tests, but that can't really be split cleanly with simple percentage ints. I'll update this PR to use relative weights with floats.